models/_sector/tokens/tokens_base_transfers.sql

```diff
@@ -0,0 +1,65 @@
+{{config(
+    schema = 'tokens',
+    alias = 'base_transfers',
+    partition_by = ['blockchain', 'token_standard', 'block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')],
+    unique_key = ['blockchain', 'unique_key']
+)
+}}
+
+{% set models = [
+    ref('tokens_arbitrum_base_transfers')
+    , ref('tokens_avalanche_c_base_transfers')
+    , ref('tokens_base_base_transfers')
+    , ref('tokens_bnb_base_transfers')
+    , ref('tokens_celo_base_transfers')
+    , ref('tokens_ethereum_base_transfers')
+    , ref('tokens_fantom_base_transfers')
+    , ref('tokens_gnosis_base_transfers')
+    , ref('tokens_optimism_base_transfers')
+    , ref('tokens_polygon_base_transfers')
+    , ref('tokens_zksync_base_transfers')
+    , ref('tokens_zora_base_transfers')
+] %}
+
+with base_union as (
```

Comment:
why do we need the base union here, can't we just have the union loop below?

Reply:
we can, yes, but i was matching the dex/nft design pattern. the difference being we apply a duplicate check on those two sectors, to help prevent loading in duplicates if a project upstream writes dupes then fails later. i left out the duplicates check here to ensure we can get it to run, but it'd be nice to have the structure in place in case we add later

Reply:
We should drop the duplicate check there as well, that should be handled by merge intos and in each model.

Reply:
the issue is on historical runs, whether full refresh due to token adds/other upstream changes or edit to model itself. sometimes we've hit duplicates on historical run when joining to prices, we get dupes. or any other bug that could have been introduced over time. this won't be handled in merge statements, as its simply an insert

Reply:
it's relatively rare to happen, but it's happened more than once in both dex/nft, so rob had ended up adding the dupe check so we don't have to fix the main sector spell